internal/core/src/segcore/SegmentInterface.cpp

```diff
@@ -86,7 +86,8 @@ SegmentInternalInterface::Search(
     query::ExecPlanNodeVisitor visitor(*this, timestamp, placeholder_group);
     auto results = std::make_unique<SearchResult>();
     *results = visitor.get_moved_result(*plan->plan_node_);
-    results->segment_ = (void*)this;
+    results->segment_ = static_cast<void*>(
+        const_cast<milvus::segcore::SegmentInternalInterface*>(this));
```

Comment:
const_cast?

Reply:
First, **const_cast** is used to remove the const-ness from `this` (which is of type `const SegmentInternalInterface `)
Then **static_cast** is used to convert the pointer to void* (which is the type of `results->segment_`)